docs/how_to_write_packages.md

```diff
@@ -1,3 +1,265 @@
+data-gatherers:
+- k8s/pods
+sections:
+  - id: images
+    name: Images
+    description: "Restrictions over the images."
+    rules:
+      - id: tag_not_latest
+        name: "Tag is not latest"
+        description: >
+          Avoid using "latest" as tag for the image since.
+        remediation: >
+          Change your manifest and edit the Pod template so the image is pinned to a certain tag.
+        links:
+          - "https://kubernetes.io/docs/concepts/containers/images/"
+```
+
+## Writing the policy definition in Rego
+
+In the previous section, we created the _policy manifest_, which contains a human readable description of the rules in our policy. Now it's time to define the same rules in a language that is machine readable.
+
+### The Rego package
+
+Preflight relies on Open Policy Agent as the policy engine. Rego is OPA's language to define policies. You can find a comprenhensive [documentation](https://www.openpolicyagent.org/docs/latest/policy-language/).
+
+You can have multiple Rego files inside the directory of a Preflight package.  All the Rego rules corresponding to the _policy manifest_ rules must be in the same Rego package, and that package must be indicated in the _policy manifest_ using the `root-query` property.
+
+For instance, this snippet shows an arbitrary Rego rule in a package named `podsbestpractices`:
+
+```
+package pods
+
+import input["k8s/pods"] as pods
+
+preflight_tag_not_latest {
+  true
+}
+```
+
+As you can identify, the Rego package for that policy is `pods`. In this case, OPA's `root-query` is `data.pods`, and that is why in the previous section, `policy-manifest.yaml` contains `root-query: "data.pods"`.
+
+### Writing Rego rules
+
+Rego can be challenging at the beginning because it does not behaves like a traditional programming language. It is strongly recommended to read ["The Basics"](https://www.openpolicyagent.org/docs/latest/policy-language/#the-basics). Also, it is useful to have the [language refence](https://www.openpolicyagent.org/docs/latest/policy-reference/) at hand.
+
+You will get faster as you write more Rego rules. In order to speed up this process, it's best to write tests for your rules, even if you think they are not needed. It means you can iterate fast while writing rules and make sure the rules are doing what you intended. It is conventional to name the test files for `policy.rego` as `policy_test.rego`.
+
+
+This example contains the definition for the `tag_no_latest` rule. As you can see, there is the convention within Preflight to add `preflight_` as prefix to the rule ID when that is written in Rego (related issue #27).
+
+```
+# preflight-packages/examples.jetstack.io/podsbestpractices/policy.rego
+
+package pods
+
+import input["k8s/pods"] as pods
+
+default preflight_tag_not_latest = false
+preflight_tag_not_latest {
+  count(containers_using_latest) == 0
+}
+
+format_container(pod, container) = name {
```

Comment:
Do we need this in the example? As I'm not sure exactly how OPA is being used this information might be returned but it seems the only 'result' is `count(containers_using_latest) == 0` in `preflight_tag_not_latest` (e.g. the container names are not made visible via that rule)

Reply:
We probably don't need this here. There are a few examples like this where a package's Rego rules give additional information in the output. When Preflight was a bash script it just printed the output of `opa eval`, but theres not yet any way to surface this extra information in the current implementation.

Reply:
:+1: Tbh - this feels like an MVP feature to me. Not knowing _which_ pods are failing my resource checks isn't very helpful.

Reply:
I thought this was needed so in case of two containers with the same name but from a different pod can appear in `all_containers`.

Reply:
I don't think this should hold up this PR now we have https://github.com/jetstack/preflight/issues/36 and https://github.com/jetstack/preflight/issues/37